scripts/tests/test_helpers.bash

```diff
@@ -0,0 +1,39 @@
+# Messaging
+debug () {
+	[ -z "$DEBUG_BATS" ] || echo "  DEBUG: $@" >&3
+}
+
+debug_run () {
+	debug "> lines=${lines[@]}"
+    debug "> STATUS=$status"
+}
+
+fail_msg () {
+	echo -e "FAIL: $@" >&3
+}
+
+fail () {
+	fail_msg "$@"
+	exit 1
+}
+
+error () {
+	echo -e "ERROR: $@" >&3
+	exit 2
+}
+
+# Assertion helpers
+
+assert_equal () {
+	if [ "$#" != 2 ]; then
+		error "Got $# arguments to eq whenever expected 2"
+	fi
+	if [ "$1" != "$2" ]; then
+		fail "Arguments are not equal.\n #1=<<$1>>\n #2=<<$2>>"
+	fi
+}
+
+assert_clean_exit () {
+	assert_equal "$status" 0
```

Comment:
In general I'm not thrilled about referencing globals, but I can see the utility here.

Reply:
yeah, me neither -- but here I liked how succinct test code then looks ;)

Reply:
Is the plan to replace these custom assert functions with the ones that duplicate in the ztombol/bats-assert library?

Reply:
I think so -- better to reuse than to rebrew ;)